src/allowed-blocks-setup.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { map, without } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { dispatch } from '@wordpress/data';
+import { getBlockTypes } from '@wordpress/blocks';
+/**
+ * Internal dependencies
+ */
+import { JETPACK_DATA_PATH } from '../jetpack/projects/plugins/jetpack/extensions/shared/get-jetpack-data';
+import { registerBlock as registerJetpackLayoutGridBlock } from '../block-experiments/blocks/layout-grid/src';
+import {
+	registerContactInfoBlock as registerJetpackContactInfoBlock,
+	registerStoryBlock as registerJetpackStoryBlock,
+} from '../jetpack/projects/plugins/jetpack/extensions/editor.native';
+
+// Please also consider updating ./block-support/supported-blocks.json
+const availableJetpackBlocks = {
+	'contact-info': { available: true },
+	'layout-grid': { available: true },
+	story: { available: true },
+};
+
+const setJetpackData = ( {
+	isJetpackActive = false,
+	userData: tracksUserData = null,
+	siteFragment = null,
+	blogId: wpcomBlogId = 1,
+} ) => {
+	return {
+		siteFragment,
+		tracksUserData,
+		wpcomBlogId,
+		jetpack: { is_active: isJetpackActive },
+		available_blocks: availableJetpackBlocks,
+	};
+};
+
+const registerJetpackBlocksIfCapable = ( props = {} ) => {
+	const {
+		capabilities: {
+			layoutGridBlock = false,
+			mediaFilesCollectionBlock = false,
+			contactInfoBlock = false,
+		} = {},
+	} = props;
+
+	if ( layoutGridBlock ) {
+		registerJetpackLayoutGridBlock();
+	}
+
+	if ( mediaFilesCollectionBlock ) {
+		registerJetpackStoryBlock();
+	}
+
+	if ( contactInfoBlock ) {
+		registerJetpackContactInfoBlock();
+	}
+};
+
+export const setupJetpackBlocks = ( props = {} ) => {
+	const { jetpackState = { blogId: 1, isJetpackActive: true } } = props;
+	const { isJetpackActive = false } = jetpackState;
+
+	global.window[ JETPACK_DATA_PATH ] = setJetpackData( jetpackState );
```

Comment:
I couldn't see any adverse side-effects of changing it because all usages of `getJetpackData` where the `JETPACK_DATA_PATH` is not set on the `global` object have reasonable default values, and the usages of the fields on the Jetpack Data are guarded.

That being said I think I just answered my own question that `wpcomBlogId` should probably remain `undefined` as a default value, and I understand the reluctance with wanting to set the field on the `global` object even if `{jetpack: { is_active: false }}` so I'll revert to not setting it at all like it was previously doing.

```
if (
    typeof window === 'object' &&
    typeof window.Jetpack_Editor_Initial_State === 'object' &&
    typeof window.Jetpack_Editor_Initial_State.tracksUserData === 'object' &&
    typeof window.Jetpack_Editor_Initial_State.wpcomBlogId !== 'undefined'
) {
```
```
if (
    window &&
    window.Jetpack_Editor_Initial_State &&
    window.Jetpack_Editor_Initial_State.siteFragment
) {
```

- [ ] Revert to not setting the `JETPACK_DATA_PATH` unless Jetpack is `active`.

Reply:
Done.